tests/client_integration_test.go

```diff
@@ -799,6 +801,96 @@ func (s *clientIntegrationSuite) TestContinueAsNewTightLoop() {
 	s.GreaterOrEqual(duration, time.Second*4)
 }
 
+func (s *clientIntegrationSuite) TestStickyAutoReset() {
+	// This test starts a workflow, wait and verify that the workflow is on sticky task queue.
+	// Then it stops the worker for 10s, this will make matching aware that sticky worker is dead.
+	// Then test sends a signal to the workflow to trigger a new workflow task.
+	// Test verify that workflow is still on sticky task queue.
+	// Then test poll the original workflow task queue directly (not via SDK),
+	// and verify that the polled WorkflowTask contains full history.
+	workflowId := "sticky_auto_reset"
+	wfFn := func(ctx workflow.Context) (string, error) {
+		sigCh := workflow.GetSignalChannel(ctx, "sig-name")
+		var msg string
+		sigCh.Receive(ctx, &msg)
+		return msg, nil
+	}
+
+	s.worker.RegisterWorkflow(wfFn)
+
+	ctx, cancel := rpc.NewContextWithTimeoutAndVersionHeaders(time.Minute)
+	defer cancel()
+	options := sdkclient.StartWorkflowOptions{
+		ID:                 workflowId,
+		TaskQueue:          s.taskQueue,
+		WorkflowRunTimeout: time.Minute,
+	}
+	// start the test workflow
+	future, err := s.sdkClient.ExecuteWorkflow(ctx, options, wfFn)
+	s.NoError(err)
+
+	// wait until wf started and sticky is set
+	var stickyQueue string
+	for i := 0; i < 5; i++ {
+		ms, err := s.adminClient.DescribeMutableState(ctx, &adminservice.DescribeMutableStateRequest{
+			Namespace: s.namespace,
+			Execution: &commonpb.WorkflowExecution{
+				WorkflowId: future.GetID(),
+			},
+		})
+
+		s.NoError(err)
+		stickyQueue = ms.DatabaseMutableState.ExecutionInfo.StickyTaskQueue
+		// verify workflow has sticky task queue
+		if stickyQueue == "" {
+			// wait until we see sticky task queue is set
+			time.Sleep(time.Second)
+			continue
+		}
```

Comment:
did you mean to put `break` after this?

Reply:
No, I need to continue checking until stickyQueue is not empty.